route-rs-packets/src/arp.rs

```diff
@@ -143,6 +143,25 @@ impl ArpFrame {
         self.frame
     }
 
+    /// Default-specific methods
+    pub fn sender_mac_addr(&self) -> Result<MacAddr, std::array::TryFromSliceError> {
```

Comment:
Let's do `use std::array::TryFromSliceError` so we don't need to fully qualify it each time.

Reply:
Is it necessary to return a result here?  Will a valid ArpFrame ever return values that would cause this to fail?
For instance, will self.sender_hardware_addr() return a slice of values that would cause MacAddr to not be able to make an MacAddr?  The only case this happens is if the slice is too short, which won't ever happen unless the underlying vector has been emptied, which isn't a case the library actually handles, as it exposes the array publicly.  

I think you should  follow the rest of the lib and return the actual value and panic if you get a slice error. (see Ipv4Packet.src_addr().  This reduces the load on our end user significantly.

Reply:
if we want to never panic, we should probably do that all at once in a separate PR.  For now I'd like to keep the API of the library consistent across each type.

Reply:
As far as I can tell, your arp_data function would actually panic during accessing the vec via the index operator.  That won't get caught because it returns a &[u8],  so the slice out of bounds error causes a panic.  Therefore, the above `Result`s will never return an error.